test/fail_compilation/retscope2.d

```diff
@@ -162,3 +162,46 @@ void foo800()
     scope Exception e;
     throw e;
 }
+
+/*************************************************/
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*************************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope2.d(1005): Error: scope variable `p` assigned to `this` with longer lifetime
+---
+*/
+
+#line 1000
+
+class C17428
```

Comment:
That's probably already covered in the many tests in `retscope.d` and `retscope2.d`, but in any case it is off topic to this particular bug fix.